components/lib/crash/src/main/java/mozilla/components/lib/crash/service/MozillaSocorroService.kt

```diff
@@ -20,54 +39,215 @@ typealias GeckoCrashReporter = CrashReporter
  *                [File a bug](https://bugzilla.mozilla.org/enter_bug.cgi?product=Socorro) if you would like to get your
  *                app added to the whitelist.
  */
+@Suppress("TooManyFunctions", "LargeClass")
 class MozillaSocorroService(
     private val applicationContext: Context,
-    private val appName: String
+    private val appName: String,
+    private val serverUrl: String = defaultServerUrl
 ) : CrashReporterService {
-    override fun report(crash: Crash.UncaughtExceptionCrash) {
-        // Not implemented currently.
+    private val logger = Logger("mozac/MozillaSocorroCrashHelperService")
+    private val startTime = System.currentTimeMillis()
+    val partSet = mutableSetOf<String>()
```

Comment:
nit: This probably does not need to be visible? Just briefly looking at this I wonder if that needs to be a property at all or whether we could just pass a new mutableSet to every method that needs it from `sendReport()`.

Reply:
Definitely should be private.   I went back and forth on this.  I wasn't sure which way was better, but since you think passing a fresh copy everytime sending a report is better I will implement it that way.  Thanks!